pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -146,11 +145,28 @@ func (bc *backendPoolTypeNodeIPConfig) ReconcileBackendPools(clusterName string,
 	lbBackendPoolID := bc.getBackendPoolID(lbName, bc.getLoadBalancerResourceGroup(), lbBackendPoolName)
 	vmSetName := bc.mapLoadBalancerNameToVMSet(lbName, clusterName)
 
-	for _, bp := range newBackendPools {
+	for i := len(newBackendPools) - 1; i >= 0; i-- {
+		bp := newBackendPools[i]
 		if strings.EqualFold(*bp.Name, lbBackendPoolName) {
-			klog.V(10).Infof("bc.ReconcileBackendPools for service (%s)(%t): lb backendpool - found wanted backendpool. not adding anything", serviceName, wantLb)
+			klog.V(10).Infof("bc.ReconcileBackendPools for service (%s): lb backendpool - found wanted backendpool. not adding anything", serviceName)
 			foundBackendPool = true
 
+			// If the LB backend pool type is configured from nodeIP or podIP
+			// to nodeIPConfiguration, we need to decouple the VM NICs from the LB
+			// before attaching nodeIPs/podIPs to the LB backend pool.
+			if bp.BackendAddressPoolPropertiesFormat != nil &&
+				(bp.BackendIPConfigurations == nil || len(*bp.BackendIPConfigurations) == 0) {
+				if err := bc.DeleteLBBackendPool(lbName, to.String(bp.Name)); err != nil {
```

Comment:
When removing IP based backend pools from an internal LB, there is a chance to see the following error message:
```
E0121 06:43:38.991938       1 azure_loadbalancer_backendpool.go:161] bc.ReconcileBackendPools for service (default/lb-svc-1-internal): failed to DeleteLBBackendPool: Retriable: true, RetryAfter: 0s, HTTPStatusCode: 0, RawError: Code="Canceled" Message="Operation was canceled." Details=[{"code":"CanceledAndSupersededDueToAnotherOperation","message":"Operation DeleteLoadBalancerBackendAddressPoolOperation (e247e03e-5eff-460f-a745-f5cd29b8d9bc) was canceled and superseded by operation DeleteLoadBalancerBackendAddressPoolOperation (15442665-d7e8-4f1d-bc9c-ee5dccd87bca)."}]
```
Suspecting this is a bug from the SLB side.

Reply:
It would retry and succeed finally.